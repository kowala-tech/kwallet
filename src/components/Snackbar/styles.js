const styles = (theme) => ({
	close: {
		width: theme.spacing.unit * 4,
		height: theme.spacing.unit * 4,
	},
});

export default styles;
