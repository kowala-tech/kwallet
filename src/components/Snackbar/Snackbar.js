import React from "react";
import PropTypes from "prop-types";
// Material UI Imports
import { withStyles } from "@material-ui/core/styles";
import MuiSnackbar from "@material-ui/core/Snackbar";
import IconButton from "@material-ui/core/IconButton";
import CloseIcon from "@material-ui/icons/Close";
// Component Related Imports
import styles from "./styles";

class Snackbar extends React.Component {
	render() {
		const {
			classes,
			open,
			handleClose,
			message
		} = this.props;

		return (
			<MuiSnackbar
				anchorOrigin={{
					vertical: "bottom",
					horizontal: "right",
				}}
				open={open}
				autoHideDuration={5000}
				onClose={handleClose}
				onExited={handleClose}
				message={<span id="message-id">{message}</span>}
				action={[
					<IconButton
						key="close"
						aria-label="Close"
						color="inherit"
						className={classes.close}
						onClick={handleClose}
					>
						<CloseIcon />
					</IconButton>
				]}
			/>
		);
	}
}

Snackbar.propTypes = {
	classes: PropTypes.object.isRequired,
	message: PropTypes.string,
	open: PropTypes.bool,
	handleClose: PropTypes.func
};

export default withStyles(styles)(Snackbar);
