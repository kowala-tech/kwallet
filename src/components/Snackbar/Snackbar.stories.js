import React from "react";
import { storiesOf } from "@storybook/react";
// import { action } from "@storybook/addon-actions";
// Story related component imports
import Snackbar from "./Snackbar";

storiesOf("Snackbar", module)
	.add("default", () =>
		(<Snackbar
			message={"Example message"}
		/>)
	);
