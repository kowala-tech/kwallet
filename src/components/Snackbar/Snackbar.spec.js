import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
// Story related component imports
import Snackbar from "./Snackbar";

describe("Snackbar", () => {

	it("it should match the snapshot", () => {
		const component = toJson(
			shallow(
				<Snackbar
					message={"Hello! I am a snackbar."}
				/>
			)
		);
		expect(component).toMatchSnapshot();
	});

});
