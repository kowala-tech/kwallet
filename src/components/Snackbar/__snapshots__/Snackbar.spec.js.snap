// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar it should match the snapshot 1`] = `
<Snackbar
  classes={
    Object {
      "close": "Snackbar-close-1",
    }
  }
  message="Hello! I am a snackbar."
/>
`;
